src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -362,6 +362,15 @@ internal override (CSharpAttributeData?, BoundAttribute?) EarlyDecodeWellKnownAt
 
             return base.EarlyDecodeWellKnownAttribute(ref arguments);
         }
+
+        /// <summary>
+        /// Binds attributes applied to this method.
+        /// </summary>
+        public ImmutableArray<(CSharpAttributeData, BoundAttribute)> BindMethodAttributes()
+        {
+            Debug.Assert(OuterBinder is not null);
+            return BindAttributes(GetAttributeDeclarations(), OuterBinder);
```

Comment:
>OuterBinder [](http://example.com/codeflow?start=62&length=11)

Why are we using `OuterBinder` rather than `WithTypeParametersBinder` here?

Reply:
That's what's used in `GetAttributesBag`, I'm just trying to be consistent.

https://github.com/dotnet/roslyn/blob/a7f80599c74692d58c41eaf9998488387dcac63a/src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs#L286

Reply:
Is there a test that will fail if someone (in future) changed `OuterBinder` to `WithTypeParametersBinder` in this code path?

What if the generic parameter is used inside `nameof(..)` which is allowed?

I think constants usage in attributes should be consistent with the following existing behavior

```csharp
using System;
public class C
{
    public void M()
    {
        // no compiler warning
        const int X = 5;
        Console.WriteLine(nameof(X));
        
        // warning CS0219: The variable 'Y' is assigned but its value is never used
        const int Y = 6;
    }
}
```

ie, `[MyAttr(nameof(UnusedConstant))]` shouldn't give a warning

Reply:
> Is there a test that will fail if someone (in future) changed `OuterBinder` to `WithTypeParametersBinder` in this code path?

No, but I can add one: dba8ea61d7a2d521225f107712c41aa3d932ddc1. It fails whenever `OuterBinder` is changed to `WithTypeParametersBinder` in either (or both) of the code paths.

> What if the generic parameter is used inside `nameof(..)` which is allowed?

That actually works even with the `OuterBinder` (not sure why, but it binds properly; although I think it's not possible to test it with the usage analysis).

Reply:
Thanks. Adding such a test would be good.
Yes, the special behavior inside nameof is expected (this was a C# 11 feature). Thanks for clarifying.